test/test_symmbasis.jl

```diff
@@ -73,7 +73,58 @@ for L = 0 : 3
       println()
 end
 
-##
+#---
+@info("SymmetricBasis construction and evaluation: Spherical Matrix")
+
+for L1 = 0:3
+   for L2 = 0:3
+
+      if L1 == 0
+         if L2 == 0 @info("Test 01-05: L1 = 0, L2 = 0 ↔ s-s block")
+         elseif L2 == 1 @info("Test 06-10: L1 = 0, L2 = 1 ↔ p-s block")
+         elseif L2 == 2 @info("Test 11-15: L1 = 0, L2 = 2 ↔ d-s block")
+         elseif L2 == 3 @info("Test 16-20: L1 = 0, L2 = 3 ↔ f-s block")
+         end
+      elseif L1 == 1
+         if L2 == 0 @info("Test 21-25: L1 = 1, L2 = 0 ↔ s-p block")
+         elseif L2 == 1 @info("Test 26-30: L1 = 1, L2 = 1 ↔ p-p block")
+         elseif L2 == 2 @info("Test 31-35: L1 = 1, L2 = 2 ↔ d-p block")
+         elseif L2 == 3 @info("Test 36-40: L1 = 1, L2 = 3 ↔ f-p block")
+         end
+      elseif L1 == 2
+         if L2 == 0 @info("Test 41-45: L1 = 2, L2 = 0 ↔ s-d block")
+         elseif L2 == 1 @info("Test 46-50: L1 = 2, L2 = 1 ↔ p-d block")
+         elseif L2 == 2 @info("Test 51-55: L1 = 2, L2 = 2 ↔ d-d block")
+         elseif L2 == 3 @info("Test 56-60: L1 = 2, L2 = 3 ↔ f-d block")
+         end
+      elseif L1 == 3
+         if L2 == 0 @info("Test 61-65: L1 = 3, L2 = 0 ↔ s-f block")
+         elseif L2 == 1 @info("Test 66-70: L1 = 3, L2 = 1 ↔ p-f block")
+         elseif L2 == 2 @info("Test 71-75: L1 = 3, L2 = 2 ↔ d-f block")
+         elseif L2 == 3 @info("Test 76-80: L1 = 3, L2 = 3 ↔ f-f block")
+         end
+      end
+
+      φ = ACE.SphericalMatrix(L1, L2; T = ComplexF64)
+      pibasis = PIBasis(B1p, ord, maxdeg; property = φ, isreal = false)
+      basis = SymmetricBasis(pibasis, φ)
+      BB = evaluate(basis, cfg)
+
+      for ntest = 1:5
+         Q, D1, D2 = ACE.Wigner.rand_QD(L1, L2)
```

Comment:
* Can you please fix `rand_QD` to do the following: return an *orthogonal* matrix `Q` with the sign already integrated, and modified D-matrices `D1, D2` with the `(-1)^L1` already integrated into `D1`.

Reply:
So the usage should end up 
```julia
Q, D1, D2 = ACE.Wigner.rand_QD(L1, L2)
Xs1 = shuffle(Ref(Q) .* Xs)
D1txBB1xD2 = Ref(D1') .* BB1 .* Ref(D2)
```
I.e. no more `rand_ref`

Reply:
please fix this also in the other test